sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js

```diff
@@ -0,0 +1,137 @@
+Ext.define('Slate.cbl.view.student.DashboardController', {
+    extend: 'Ext.app.ViewController',
+    alias: 'controller.slate-cbl-student-dashboard',
+    requires: [
+        'Slate.cbl.model.Student',
+        'Slate.cbl.model.Competency',
+        'Slate.cbl.view.student.skill.OverviewWindow'
+    ],
+
+    config: {
+        id: 'slate-cbl-teacher-dashboard', // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+        control: {
+            '#': {
+//                contentareachange: 'refresh',
+//                progressrowclick: 'onProgressRowClick',
+                democellclick: 'onDemoCellClick',
+                render: 'onComponentRender'
+            }
+        }
+    },
+    
+    // lifecycle overrides
+    init: function() {
+        var me = this;
+
+        Ext.create('Ext.data.Store', {
+            storeId: 'cbl-competencies-loaded',
+            model: 'Slate.cbl.model.Competency'
+        });
+    },
+
+
+    //event handlers
+    onComponentRender: function(dashboardView) {
+        var me = this,
+            siteEnv = window.SiteEnvironment || {},
+            student = siteEnv.cblStudent && Ext.create('Slate.cbl.model.Student', siteEnv.cblStudent)
+            studentDashboardCompetenciesList = dashboardView.el.down('#studentDashboardCompetenciesList'),
+            contentArea = siteEnv.cblContentArea && Ext.create('Slate.cbl.model.ContentArea', siteEnv.cblContentArea),
+            competenciesStore = Ext.getStore('cbl-competencies-loaded'),
+            competenciesTpl = Ext.XTemplate.getTpl(me.view, 'competenciesTpl');
+
+        if (!student || !contentArea) {
+            return;
+        }
+
+        // empty competencies list
+        studentDashboardCompetenciesList.empty();
+        studentDashboardCompetenciesList.removeCls('competencies-unloaded').addCls('competencies-loading');
+
+        contentArea.getCompetenciesForStudents([student.getId()], function(competencies) {
+            var competenciesLength = competencies.length,
+                competencyIndex = 0,
+                competency, skillsList;
+            
+
+            if(!competenciesStore.isLoaded()) {
+                competenciesStore.loadRawData(competencies);
+            }
+
+            competenciesTpl.overwrite(studentDashboardCompetenciesList, {
+                student: student.getData(),
+                competencies: competencies
+            });
+
+            studentDashboardCompetenciesList.removeCls('competencies-loading').addCls('competencies-loaded');
+
+            for (; competencyIndex < competenciesLength; competencyIndex++) {
+                competency = Ext.create('Slate.cbl.model.Competency', competencies[competencyIndex]);
+                skillsList = studentDashboardCompetenciesList.down('.cbl-competency-panel[data-competency="'+competency.getId()+'"] .cbl-skill-meter');     
```

Comment:
Well this does all the DOM reads in the loop, and currently all the writes happen later after the async loads. DOM operations in a loop are OK as long as reads and writes are batched together which they are here. I'm not totally happy with all the async calls happening to populate this screen but its design seemed to be in flux still and I needed to get an initial version out the door.